kasa/smart/smartdevice.py

```diff
@@ -162,17 +166,34 @@ async def _initialize_modules(self):
         """Initialize modules based on component negotiation response."""
         from .smartmodule import SmartModule
 
+        # Some of the wall switches (1 at time of writing, ks240) have child
+        # devices and report the child's components on the parent, even though
+        # they don't work on the parent directly. This logic below
+        # is ensuring that WallSwitches with children report all the child
+        # modules at the parent level except for a whitelist of modules that
+        # that are reported at the child level but only work on the parent,
+        # i.e. time, firmware etc.
```

Comment:
```suggestion
        # Some wall switches (like ks240) are internally presented as having child
        # devices which report the child's components on the parent's sysinfo, even
        # when they need to be accessed through the children.
        # The logic below ensures that such devices report all but whitelisted the child
        # modules at the parent level to create an illusion of a single device.
```
Maybe something shorter like this?

Reply:
Done